anndata/tests/test_awkward.py

```diff
@@ -0,0 +1,284 @@
+def test_copy(key):
+    """Check that modifying a copy does not modify the original"""
+    adata = gen_adata((3, 3), varm_types=(), obsm_types=(), layers_types=())
+    getattr(adata, key)["awk"] = ak.Array([{"a": [1], "b": [2], "c": [3]}] * 3)
+    adata_copy = adata.copy()
+    getattr(adata_copy, key)["awk"]["c"] = np.full((3, 1), 4)
+    getattr(adata_copy, key)["awk"]["d"] = np.full((3, 1), 5)
+
+    # values in copy were correctly set
+    npt.assert_equal(getattr(adata_copy, key)["awk"]["c"], np.full((3, 1), 4))
+    npt.assert_equal(getattr(adata_copy, key)["awk"]["d"], np.full((3, 1), 5))
+
+    # values in original were not updated
+    npt.assert_equal(getattr(adata, key)["awk"]["c"], np.full((3, 1), 3))
+    with pytest.raises(IndexError):
+        getattr(adata, key)["awk"]["d"]
+
+
+@pytest.mark.parametrize("key", ["obsm", "varm"])
+def test_view(key):
+    """Check that modifying a view does not modify the original"""
+    adata = gen_adata((3, 3), varm_types=(), obsm_types=(), layers_types=())
+    getattr(adata, key)["awk"] = ak.Array([{"a": [1], "b": [2], "c": [3]}] * 3)
+    adata_view = adata[:2, :2]
+
+    with pytest.warns(ImplicitModificationWarning, match="initializing view as actual"):
+        getattr(adata_view, key)["awk"]["c"] = np.full((2, 1), 4)
+        getattr(adata_view, key)["awk"]["d"] = np.full((2, 1), 5)
+
+    # values in view were correctly set
+    npt.assert_equal(getattr(adata_view, key)["awk"]["c"], np.full((2, 1), 4))
+    npt.assert_equal(getattr(adata_view, key)["awk"]["d"], np.full((2, 1), 5))
+
+    # values in original were not updated
+    npt.assert_equal(getattr(adata, key)["awk"]["c"], np.full((3, 1), 3))
+    with pytest.raises(IndexError):
+        getattr(adata, key)["awk"]["d"]
+
+
+@pytest.mark.parametrize(
+    "array",
+    [
+        # numpy array
+        ak.Array(np.arange(2 * 3 * 4 * 5).reshape((2, 3, 4, 5))),
+        # record
+        ak.Array([{"a": 1, "b": 2}, {"a": 1, "b": 3}]),
+        # ListType, variable length
+        ak.Array([[1], [2, 3], [4, 5, 6]]),
+        # RegularType + nested ListType
+        ak.to_regular(ak.Array([[[1, 2], [3]], [[2], [3, 4, 5]]]), 1),
+        # nested record
+        ak.to_regular(ak.Array([[{"a": 0}, {"b": 1}], [{"c": 2}, {"d": 3}]]), 1),
+        # mixed types (variable length)
+        ak.Array([[1, 2], ["a"]]),
+        # zero-size edge cases
+        ak.Array(np.ones((0, 7))),
+        ak.Array(np.ones((7, 0))),
+        # UnionType of two regular types with different dimensions
+        ak.concatenate([ak.Array(np.ones((2, 2))), ak.Array(np.ones((2, 3)))]),
+        # UnionType of two regular types with same dimension
+        ak.concatenate(
+            [
+                ak.Array(np.ones((2, 2))),
+                ak.Array(np.array([["a", "a"], ["a", "a"]])),
+            ]
+        ),
+        # categorical array
+        ak.operations.to_categorical(ak.Array([["a", "b", "c"], ["a", "b"]])),
```

Comment:
Updating syntax and adding case for non-string categoricals

```suggestion
        ak.to_categorical(ak.Array([["a", "b", "c"], ["a", "b"]])),
        ak.to_categorical(ak.Array([[1, 1, 2], [3, 3]])),
```